src/internal/channel.js

```diff
@@ -113,7 +113,7 @@ export function channel(buffer) {
 }
 
 export function eventChannel(subscribe, buffer = buffers.none(), matcher) {
-  if(arguments.length > 1)
+  if(typeof matcher !== 'undefined')
```

Comment:
IMO it's be better to write `if(arguments.length > 2)`. because this would allows catching common typo Errors: i.e. the user provided a matcher argument but provided arg is undefined due to some typo error.